### PR TITLE
[14.0][NEW] hr_timesheet_time_restriction

### PR DESCRIPTION
**Pull request by GabbasovDinar**

### CODE REVIEW

**Review by francesco-ooops — APPROVED** [review]
From description: 

This app restricts user from creating timesheets for past dates (or changing date of timesheet to a past date) based on project configuration or global restriction settings.

Use case is to make sure that new timesheets are not added to a specific time range after that period has been reported and/or billed.

In user, it's possible to activate flag "No timesheet date restriction" to override behavior.

**Review by elvise — APPROVED** [review]
LGTM!

**Comment by francesco-ooops** [comment]
@OCA/human-resources-maintainers could you review? thanks!

**Comment by francesco-ooops** [comment]
@OCA/human-resources-maintainers weekly request for review :)

File: hr_timesheet_time_restriction/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2022 Dinar Gabbasov
+# Copyright 2022 Ooops404
+# Copyright 2022 Cetmix
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Time Restriction",
```
Comment (rafaelbn) [inline hr_timesheet_time_restriction/__manifest__.py]: HR Timesheet Sheet Restriction

Recomended, ambituous names are problematic, time restriction where? Here in the PR you have context but in OCA Apps, Odoo APPs o inside Odoo in APs the name is more relevant.

File: hr_timesheet_time_restriction/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "version": "14.0.1.0.0",
+    "category": "Human Resources",
+    "website": "https://github.com/OCA/timesheet",
+    "author": "Cetmix, Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "installable": True,
+    "application": False,
+    "summary": "Time Restriction",
```
Comment (rafaelbn) [inline hr_timesheet_time_restriction/__manifest__.py]: Is this a summary?

**Review by rafaelbn — COMMENTED** [review]
I'm not PSC here but this can be done using timesheet sheets so I won't use is. Sorry.

In any case, thank you for you contribution! It's great and I wish you find reviewers.

Usually, screenshots of 30 second videos help because a much easier is the review more reviews.

My best regards

**Comment by francesco-ooops** [comment]
> I'm not PSC here but this can be done using timesheet sheets so I won't use is. Sorry.

Hi @rafaelbn , thanks for your feedback. Your comments are correct.

Could you explain how to achieve the same result as this module with timesheet sheet?

**Comment by rafaelbn** [comment]
As I remember, if the timesheet responsible validate  the timesheet **sheet** of an employee, the employee cannot modify timesheets any more. So responsible should create and validate timesheet sheet every week or every month. I wish this is also in v14 I remember it from v12.

**Comment by francesco-ooops** [comment]
> As I remember, if the timesheet responsible validate the timesheet **sheet** of an employee, the employee cannot modify timesheets any more. So responsible should create and validate timesheet sheet every week or every month. I wish this is also in v14 I remember it from v12.

@rafaelbn I understand what you mean, I think what you propose would work better for weekly timesheet sheets and a bit less for daily ones, and it would also require a daily task from timesheet manager/HR so I think it's good to have multiple options by adding also this module :)

Your comments have been addressed, would you approve now? :)

**Comment by elvise** [comment]
> As I remember, if the timesheet responsible validate the timesheet **sheet** of an employee, the employee cannot modify timesheets any more. So responsible should create and validate timesheet sheet every week or every month. I wish this is also in v14 I remember it from v12.

Thanks @rafaelbn for your comment.
With the "timesheet sheet" it's not possible to modify a validated "timesheet sheet", but it is still possible to insert timesheets with a date antecedent to the creation date and if you work with weekly billing this is a big problem.
Let check this scenario :
Employee put the timesheet on monday for the previous week, but sunday you have sent the invoice to the customer.

Let me know if it's more clear now

**Comment by elvise** [comment]
@pedrobaeza why not merge also this one 👍

**Comment by pedrobaeza** [comment]
As there are comments from @rafaelbn, he should say if everything is correct according him.

**Review by rafaelbn — APPROVED** [review]
My comments of name and summary have been attended 👍

**Pull request merged**